taxi/views.py

```diff
@@ -36,6 +43,21 @@ class ManufacturerListView(LoginRequiredMixin, generic.ListView):
     template_name = "taxi/manufacturer_list.html"
     paginate_by = 5
 
+    def get_context_data(self, *, object_list=None, **kwargs):
+        context = super(ManufacturerListView, self).get_context_data(**kwargs)
+        search_name = self.request.GET.get("name", "")
+        context["search_form"] = ManufacturerSearchForm(
+            initial={"name": search_name}
+        )
+        return context
+
+    def get_queryset(self):
+        queryset = Manufacturer.objects.all()
+        search_name = self.request.GET.get("name")
+        if search_name:
+            return queryset.filter(name__icontains=search_name)
+        return queryset
```

Comment:
You should use `ManufacturerNameSearchForm` to validate your input here

```
form = ManufacturerNameSearchForm(self.request.GET)
    if form.is_valid():
```

And then filter queryset with this name